### PR TITLE
MULE-17064: Application Fails when using colon (:) character in SFTP Directory path on Windows only

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

File: src/main/java/org/mule/extension/sftp/internal/command/SftpReadCommand.java [hunk 1]
```diff
@@ -64,9 +65,9 @@ public SftpReadCommand(SftpFileSystem fileSystem, SftpClient client) {
   @Override
   public Result<InputStream, SftpFileAttributes> read(FileConnectorConfig config, SftpFileAttributes attributes, boolean lock,
                                                       Long timeBetweenSizeCheck) {
-    Path path = Paths.get(attributes.getPath());
+    URI uri = URI.create(attributes.getPath());
```
Comment (ndinu) [inline src/main/java/org/mule/extension/sftp/internal/command/SftpReadCommand.java]: Why not use UriUtils?

File: src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java [hunk 2]
```diff
@@ -317,8 +317,10 @@ public boolean isConnected() {
       return emptyList();
     }
 
-    return entries.stream().map(entry -> new SftpFileAttributes(Paths.get(path).resolve(entry.getFilename()), entry.getAttrs()))
+    List list3 = entries.stream().map(entry -> new SftpFileAttributes(createUri(path, entry.getFilename()), entry.getAttrs()))
```
[the remark this answers is not in the document]
Comment (SebaElizalde) [reply src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java]: Fixed. It was named like that for testing purposes and then the name slipped through.

File: src/main/java/org/mule/extension/sftp/internal/AbstractSftpCopyDelegate.java [hunk 3]
```diff
@@ -60,18 +59,18 @@ public void doCopy(FileConnectorConfig config, FileAttributes source, Path targe
     } catch (ConnectionException e) {
       throw command
           .exception(format("FTP Copy operations require the use of two FTP connections. An exception was found trying to obtain second connection to"
-              + "copy the path '%s' to '%s'", source.getPath(), targetPath), e);
+              + "copy the path '%s' to '%s'", source.getPath(), targetUri), e);
```
[the remark this answers is not in the document]
Comment (SebaElizalde) [reply src/main/java/org/mule/extension/sftp/internal/AbstractSftpCopyDelegate.java]: Agreed. Seems like I missed a couple of them, fixing them now.

File: src/main/java/org/mule/extension/sftp/internal/command/SftpReadCommand.java (hunk 1 above)
Comment (SebaElizalde) [reply src/main/java/org/mule/extension/sftp/internal/command/SftpReadCommand.java]: Fixed.

**Pull request merged**